sdk/identity/identity/src/util/validateMultiTenant.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetTokenOptions } from "@azure/core-auth";
+
+/**
+ * @internal
+ */
+export const multiTenantError = new Error(
+  "Multi-tenant authentication was attempted, but multi-tenant authentication was not enabled in this credential instance."
+);
+
+/**
+ * Verifies whether locally assigned tenants are equal to tenants received through getToken.
+ * Returns the appropriate tenant.
+ * @internal
+ */
+export function processMultiTenantRequest(
+  tenantId?: string,
+  allowMultiTenantAuthentication?: boolean,
+  getTokenOptions?: GetTokenOptions
+): string | undefined {
+  if (
+    !allowMultiTenantAuthentication &&
+    getTokenOptions?.tenantId &&
+    tenantId &&
+    getTokenOptions.tenantId !== tenantId
+  ) {
+    throw multiTenantError;
```

Comment:
The error seems misleading, since there are two other checks besides `allowMultiTenantAuthentication` being falsy. The tenantId being missing or not matching what is in getTokenOptions seem wildly different, but wouldn't be mentioned in the error message at all

Reply:
What do you think of this? `A getToken request was attempted with a tenant different than the tenant configured at the initialization of the credential, but multi-tenant authentication was not enabled in this credential instance.` I’ll push it in the mean time.

Reply:
Within this conditional specifically, if `allowMultiTenantAuthentication` is true, none of the other checks are taken in consideration.

Reply:
This is what I get for rushing to review before a meeting 🤣